infra/aws/terraform/prow-build-cluster/resources/test-pods/externalsecrets.yaml

```diff
@@ -4,78 +4,114 @@
-#     name: credentials                                                                      # The key to write to in the Kubernetes Secret
-#     version: latest                                                                        # The version of the GSM Secret
-#     property: credentials
-# ---
-# apiVersion: kubernetes-client.io/v1
-# kind: ExternalSecret
-# metadata:
-#   name: aws-credentials-607362164682           # The name of the Kubernetes Secret
-#   namespace: test-pods
-# spec:
-#   backendType: gcpSecretsManager
-#   projectId: k8s-infra-prow-build
-#   data:
-#   - key: gke_k8s-prow-builds_us-central1-f_prow__test-pods__aws-credentials-607362164682   # The name of the GSM Secret
-#     name: credentials                                                                      # The key to write to in the Kubernetes Secret
-#     version: latest                                                                        # The version of the GSM Secret
-#     property: credentials
-# ---
-# apiVersion: kubernetes-client.io/v1
-# kind: ExternalSecret
-# metadata:
-#   name: aws-ssh-key-secret           # The name of the Kubernetes Secret
-#   namespace: test-pods
-# spec:
-#   backendType: gcpSecretsManager
-#   projectId: k8s-infra-prow-build
-#   data:
-#   - key: gke_k8s-prow-builds_us-central1-f_prow__test-pods__aws-ssh-key-secret   # The name of the GSM Secret
-#     name: aws-ssh-private                                                        # The key to write to in the Kubernetes Secret
-#     version: latest                                                              # The version of the GSM Secret
-#     property: aws-ssh-private
-#   - key: gke_k8s-prow-builds_us-central1-f_prow__test-pods__aws-ssh-key-secret   # The name of the GSM Secret
-#     name: aws-ssh-public                                                         # The key to write to in the Kubernetes Secret
-#     version: latest                                                              # The version of the GSM Secret
-#     property: aws-ssh-public
+---
+apiVersion: external-secrets.io/v1beta1
+kind: ClusterSecretStore
+metadata:
+  name: k8s-infra-prow-build
+spec:
+  provider:
+    gcpsm:
+      projectID: k8s-infra-prow-build
+---
+apiVersion: external-secrets.io/v1beta1
+kind: ExternalSecret
+metadata:
+  name: service-account
+  namespace: test-pods
+spec:
+  secretStoreRef:
+    kind: ClusterSecretStore
+    name: k8s-infra-prow-build
+  target:
+    name: service-account # The name of the Kubernetes Secret
+    creationPolicy: Owner
+  data:
+  - secretKey: service-account.json # The key to write to in the Kubernetes Secret
+    remoteRef:
+      key: prow-build-service-account # The name of the GSM Secret
+      version: latest                 # The version of the GSM Secret    
+---
+apiVersion: external-secrets.io/v1beta1
+kind: ExternalSecret
+metadata:
+  name: ssh-key-secret
```

Comment:
We don't need this SSH key. This is needed for e2e tests on GCP. we may need to create a new key pair on EC2.